.github/workflows/build-test-actions.yml

```diff
@@ -0,0 +1,44 @@
+name: build-test-actions
+
+on: [push]
+
+jobs:
+  build-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # Add versions here to expand support matrix
+        python-version: ["3.9"]
```

Comment:
It's great to check Python 3.9, however I wonder if we should also include 3.10 and 3.11 (especially since 3.11 is pinned in the `pyproject.toml` file.

Reply:
I pushed a commit adding these versions to the matrix. Let's watch and see how it goes 👀

Reply:
Looks like Python 3.10 has at least one problem, which you can see here: https://github.com/ryanprior/toqito/actions/runs/5117092849/jobs/9199863846

So this validates the usefulness of adding more versions to the support matrix, such that we find these problems. Do you prefer to merge this PR now with a wider matrix and failing pipeline though, or merge a green pipeline first and update the matrix later after build issues have been fixed?

Reply:
Hmm, I wonder if this is a `mypy` issue?
https://github.com/pre-commit/pre-commit/issues/2850

Preferably, I'd prefer to resolve this prior to merging if possible. Reading through the issue, it looks like we might be able to resolve the 3.11 failure by pinning to a specific `mypy` as the OP mentioned.

Reply:
Great link. [Quick bump to `mypy`](https://github.com/vprusso/toqito/commit/e2d0a92c0aa1523075bf74fbde71181344aec1b0) and we are [all green](https://github.com/ryanprior/toqito/actions/runs/5122486308).